src/containers/Application/ApplicationSummary.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react'
+import { FullStructure, ResponsesByCode, ElementStateNEW } from '../../utils/types'
+import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
+import { ApplicationStatus } from '../../utils/generated/graphql'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { Loading, NoMatch } from '../../components'
+import { SectionWrapper } from '../../components/Application'
+import strings from '../../utils/constants'
+import { Button, Header, Message, Container } from 'semantic-ui-react'
+
+interface ApplicationProps {
+  structure: FullStructure
+}
+
+const ApplicationSummary: React.FC<ApplicationProps> = ({ structure }) => {
+  const { replace } = useRouter()
+  const { error, fullStructure } = useGetFullApplicationStructure({
+    structure,
+  })
+
+  useEffect(() => {
+    if (!fullStructure) return
+    // Re-direct if application is not valid
+    if (fullStructure.info.firstStrictInvalidPage) {
+      const { sectionCode, pageNumber } = fullStructure.info.firstStrictInvalidPage
+      replace(`/applicationNEW/${fullStructure.info.serial}/${sectionCode}/Page${pageNumber}`)
+    }
+  }, [fullStructure])
+
+  const handleSubmit = () => {}
+
+  console.log('fullStructure', fullStructure)
+
+  if (error) return <Message error header={strings.ERROR_APPLICATION_PAGE} list={[error]} />
+  if (!fullStructure) return <Loading />
+  const { sections, responsesByCode, info } = fullStructure
+  return (
+    <Container>
+      <Header as="h1" content={strings.TITLE_APPLICATION_SUBMIT} />
+      {Object.values(sections).map((section) => (
+        <SectionWrapper
+          key={`ApplicationSection_${section.details.code}`}
```

Comment:
Actually we should use all keys as id vs codes, it's pretty easy to misconfigure and cause UI errors

Reply:
Sure, have changed to id 48cdb5b
Not entirely clear why it's better -- codes should always been unique per template. Are you saying it might be left blank or something?

Reply:
I noticed in by looking at summary page and noticing both section headers were the same